crates/evm/src/executor/fork/cache.rs

```diff
@@ -270,7 +265,7 @@ impl MemDb {
                 accounts.insert(add, acc.info);
 
                 let acc_storage = storage.entry(add).or_default();
-                if acc.status.contains(AccountStatus::Created) {
```

Comment:
While this should be OK, it seems that the `commit` function might need some tweaking, as it seems it wasn't recording the accounts as it should. We tried matching revm's commit fn but the issues still persisted.